current/advanced/block-import.md

```diff
@@ -0,0 +1,90 @@
+---
+slug: block-import
+lang: en
+title: The Block Import Pipeline
+---
+
+## The Import Queue
+
+The import queue is an abstract worker queue present in every Substrate node. It is not part of the
+runtime. The Import Queue is responsible for processing pieces of incoming information, verifying
+them, and if they are valid, importing them into the node's state. The most fundamental piece of
+information that the import queue processes is blocks themselves, but it is also responsible for
+importing consensus-related messages such as Justifications and, in light clients, finality proofs.
+
+The import queue collects incoming blocks from the network and stores them in a pool. The blocks are
+later checked for validity and discarded if they are not valid. Blocks that are valid are then
+imported into the node's local state.
+
+The import queue is codified abstractly in Substrate by means of the
+[`ImportQueue` trait](https://substrate.dev/rustdocs/v2.0.0-alpha.8/sp_consensus/import_queue/trait.ImportQueue.html).
+The use of a trait allows each consensus engine to provide its own specialized implementation of the
+Import Queue which may take advantage of optimization opportunities such as verifying multiple
+blocks in parallel as they come in across the network.
+
+The import queue also provides some hooks via the
+[`Link` trait](https://crates.parity.io/sp_consensus/import_queue/trait.Link.html) that can be used
+to follow its progress. A typical consumer of these hooks is the sync module which uses data from
```

Comment:
What is the sync module? Can you link to more information about it? I'm assuming it's not a specific thing or else it would the Sync Module. Also, remember that "which" should almost always be preceded by a comma ("A typical consumer of these hooks is the sync module, which uses data..."). If that doesn't feel appropriate, it's usually better to use "that" instead of "which".

Reply:
Comma addressed.

I don't really know what the sync module is. TBH I copied that from @andresilva 's review. The only place I see these methods being called is https://github.com/paritytech/substrate/blob/master/client/network/src/service.rs (and tests).

Reply:
It may be best to omit reference to the sync module if it introduces more questions than answers.